library/std/src/net/ip.rs

```diff
@@ -1488,6 +1488,35 @@ impl Ipv6Addr {
         (self.segments()[0] & 0xff00) == 0xff00
     }
 
+    /// Converts this address to an [IPv4 address]. Returns [`None`] if this address is
+    /// not IPv4-mapped.
+    ///
+    /// ::ffff:a.b.c.d become a.b.c.d
```

Comment:
The docs could use some polish and links to the RFC. Something like this maybe?

```suggestion
    /// Extracts the IPv4 address if this address is an ["IPv4-Mapped IPv6 Address" 
    /// according to IETF RFC 4291](https://tools.ietf.org/html/rfc4291#section-2.5.5.2).  
    /// Returns [`None`] otherwise.
    ///
    /// For example, `::ffff:a.b.c.d` becomes `a.b.c.d`. All addresses *not* starting with 
    /// `0000:0000:0000:0000:0000:ffff` will return `None`.
```

Reply:
The document has been updated.